libcnb-cargo/src/cross_compile.rs

```diff
@@ -14,52 +29,76 @@ use which::which;
 pub fn cross_compile_env(
     target_triple: impl AsRef<str>,
 ) -> Result<Vec<(OsString, OsString)>, CrossCompileError> {
-    if target_triple.as_ref() == "x86_64-unknown-linux-musl" && cfg!(target_os = "macos") {
-        let ld_binary_name = "x86_64-linux-musl-ld";
-        let cc_binary_name = "x86_64-linux-musl-gcc";
-
-        Ok(vec![
-            (
-                OsString::from("CARGO_TARGET_X86_64_UNKNOWN_LINUX_MUSL_LINKER"),
-                which(ld_binary_name)
-                    .map_err(|_| {
-                        CrossCompileError::CouldNotFindRequiredBinary(String::from(ld_binary_name))
-                    })?
-                    .into_os_string(),
-            ),
-            (
-                OsString::from("CC_x86_64_unknown_linux_musl"),
-                which(cc_binary_name)
-                    .map_err(|_| {
-                        CrossCompileError::CouldNotFindRequiredBinary(String::from(cc_binary_name))
-                    })?
-                    .into_os_string(),
-            ),
-        ])
-    } else {
-        Ok(vec![])
+    if target_triple.as_ref() == MUSL_TARGET {
+        if cfg!(target_os = "macos") {
+            return Ok(vec![
+                (
+                    OsString::from("CARGO_TARGET_X86_64_UNKNOWN_LINUX_MUSL_LINKER"),
+                    MAC_BINARIES.ld.which()?,
+                ),
+                (
+                    OsString::from("CC_x86_64_unknown_linux_musl"),
+                    MAC_BINARIES.cc.which()?,
+                ),
+            ]);
+        } else if cfg!(target_os = "linux") {
+            // Do we want to set the env vars here?
```

Comment:
The env vars do not need to be set on Linux, since cargo/rustc correctly finds the compiler there.
Ideally in the future it will be able to do so for the OS X musl-cross too:
https://github.com/rust-lang/cargo/issues/4133

As such I think we shouldn't set env vars on Linux, since I see them as more of a workaround, rather than necessary configuration.

Reply:
@edmorley thoughts on just keeping the `which` checks?